### PR TITLE
Sanntis

### PR DESCRIPTION
**Pull request by tgurbich**
Added a SanntiS module

### CODE REVIEW

File: modules/ebi-metagenomics/sanntis/meta.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  - interproscan_output:
+      type: file
+      description: Output of InterProScan in the TSV or GFF3 format.
+      pattern: "*.{tsv,gff,gff3}"
+
+  - gbk:
+      type: file
+      description: A file in .gbk format specifying the coding sequences.
+      pattern: "*.{gbk}"
```
Comment (ochkalova) [inline modules/ebi-metagenomics/sanntis/meta.yml:33]: I think {} are not required in here

**Comment by ochkalova** [comment]
I also think the snapshot should be there as well. 
No other comments, I've run nf-test on my laptop, it's fine

File: modules/ebi-metagenomics/sanntis/main.nf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    grep -v "/protein_id=" ${gbk} > ${prefix}_prepped.gbk
+    sanntis \
+    --ip-file ${interproscan_output} \
```
Comment (vagkaratzas) [inline modules/ebi-metagenomics/sanntis/main.nf:26]: push these next 5 lines after sanntis one tab inside

File: modules/ebi-metagenomics/sanntis/main.nf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'biocontainers/sanntis:0.9.3.5--pyhdfd78af_1' :
+        'biocontainers/sanntis:0.9.3.5--pyhdfd78af_1' }"
+
+    input:
+    tuple val(meta), path(interproscan_output), path(gbk)
+
+    output:
+    tuple val(meta), path("*_sanntis.gff"), emit: gff
+    path "versions.yml"           , emit: versions
```
Comment (vagkaratzas) [inline modules/ebi-metagenomics/sanntis/main.nf:15]: comma under comma :D

File: modules/ebi-metagenomics/sanntis/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
+
+    }
+
```
Comment (vagkaratzas) [inline modules/ebi-metagenomics/sanntis/tests/main.nf.test:34]: add a stub test please, check https://github.com/nf-core/modules/blob/master/modules/nf-core/goat/taxonsearch/tests/main.nf.test last test

**Review by vagkaratzas — CHANGES_REQUESTED** [review]
I am very happy with this :)
As Sonya recommended it needs the snapshot, but I understand you can't run the test locally. So, after you finish with the rest of the changes, I will push the final snapshot (it works).

File: modules/ebi-metagenomics/sanntis/meta.yml (hunk 1 above)
Comment (tgurbich) [reply modules/ebi-metagenomics/sanntis/meta.yml]: Good catch, thanks! Fixed in https://github.com/EBI-Metagenomics/nf-modules/commit/3b3d960793e4a650cc04717e253d6f6a846e2021

File: modules/ebi-metagenomics/sanntis/main.nf (hunk 2 above)
Comment (tgurbich) [reply modules/ebi-metagenomics/sanntis/main.nf]: Done in https://github.com/EBI-Metagenomics/nf-modules/commit/ad6b4bbca5144c2f7659e3d4b8c1d3e1c648f5b4

File: modules/ebi-metagenomics/sanntis/main.nf (hunk 3 above)
Comment (tgurbich) [reply modules/ebi-metagenomics/sanntis/main.nf]: Done in https://github.com/EBI-Metagenomics/nf-modules/commit/ad6b4bbca5144c2f7659e3d4b8c1d3e1c648f5b4

File: modules/ebi-metagenomics/sanntis/tests/main.nf.test (hunk 4 above)
Comment (tgurbich) [reply modules/ebi-metagenomics/sanntis/tests/main.nf.test:34]: Added in https://github.com/EBI-Metagenomics/nf-modules/commit/bbec9cfd033abab09713d17248ee60e697d91da0, hope that's how it's supposed to be done.

**Review by vagkaratzas — APPROVED** [review]
All good! Snapshot added

**Pull request merged**